mesonbuild/interpreter/type_checking.py

```diff
@@ -472,6 +472,12 @@ def link_whole_validator(values: T.List[T.Union[StaticLibrary, CustomTarget, Cus
 _ALL_TARGET_KWS: T.List[KwargInfo] = [
     KwargInfo('build_by_default', bool, default=True, since='0.40.0'),
     DEPENDENCIES_KW,
+    KwargInfo(
+        'extra_files',
+        ContainerTypeInfo(list, (str, File)),
```

Comment:
This works because build.py has an assertion (great error handler incidentally, no InvalidArguments used!) that all extra_files are of type File. It wasn't immediately obvious to me that built files weren't valid here.

Reply:
Oh, randomly looked at this because of another reported issue, and we actually added File support in 108dac5c162eb086b8cce3f4d0575f7282c72c48 (0.41.0).